pyfr/integrators/writers.py

```diff
@@ -117,31 +153,6 @@ def _write(self, path, solnmap, metadata):
             # Create the output dictionary
             outdict = dict(zip(names, solns), **metadata)
 
-            with open(path, 'wb') as f:
-                np.savez(f, **outdict)
-
-
-class DirWriter(BaseWriter):
-    writer_name = 'pyfrs-dir'
-
-    def _write(self, path, solnmap, metadata):
-        comm, rank, root = get_comm_rank_root()
-
-        # Create the output directory and save the config/status files
-        if rank == root:
-            if os.path.exists(path):
-                rm(path)
-
-            os.mkdir(path)
-
-            # Write out our metadata
-            for name, data in metadata.items():
-                np.save(os.path.join(path, name), data)
-
-        # Wait for this to complete
-        comm.barrier()
-
-        # Save the solutions
-        for etype, buf in solnmap.items():
-            solnpath = os.path.join(path, self._get_name_for_soln(etype))
-            np.save(solnpath, buf)
+            with h5py.File(path, 'w') as h5file:
+                for k, v in outdict.items():
+                    h5file.create_dataset(k, data=outdict[k])
```

Comment:
No idea what is going on here.  We're using `.items()` to get the key and the value but then completely ignoring the value and instead doing `outdict[k]`.